### PR TITLE
refactoring project

### PR DESCRIPTION
**Pull request by shamirShahzad**
Moved many files from lib to lib/iputils to remove import cycles could not use lib/ip as in the code ip.* was clashing with variables so used ipUtils instead many other files such as bogon, cidr, utils_input, init, regex, string_slice, asn_str, u128  and errors file needed to be moved as well otherwise import cycle was being created at ipinfo/cli/lib the go install ./ipinfo is not working as there is a clash with github.com/ipinfo/mmdbctl/lib/cmd_read.go  where there is use of an ip_list file which has been moved

### CODE REVIEW

**Comment by UmanShahzad** [comment]
Can we rename to `iputil` without the caps? Would match stuff like `ioutil` that is an official Go package, for example.

**Comment by shamirShahzad** [comment]
sure i will add that and create a pull request on mmdbctl which should be just one file so that i can install the ipinfo with go install ./ipinfo to test whether everthing is working can you confirm that one @UmanShahzad

File: lib/ipUtils/regex.go [hunk 1]
```diff
@@ -1,37 +1,37 @@
...
-var v4SubnetRegex *regexp.Regexp
-var v6SubnetRegex *regexp.Regexp
-var subnetRegex *regexp.Regexp
+var V4SubnetRegex *regexp.Regexp
+var V6SubnetRegex *regexp.Regexp
+var SubnetRegex *regexp.Regexp
 
-const v4octet = `(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)`
-const IPv4RegexPattern = `(` + v4octet + `\.){3}` + v4octet
+const V4octet = `(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)`
```
Comment (UmanShahzad) [inline lib/ipUtils/regex.go:33]: Capital `o`

File: lib/ipUtils/cidr.go [hunk 2]
```diff
@@ -48,10 +48,10 @@ func (c *CIDR) Size() int {
 }
 
 // list returns a slice of sorted CIDR structures.
```
Comment (UmanShahzad) [inline lib/ipUtils/cidr.go:50]: All these comments need to be fixed to capitalize the function. All!

File: lib/ipUtils/cidr.go [hunk 3]
```diff
@@ -48,10 +48,10 @@ func (c *CIDR) Size() int {
...
-func list(s []string) []*CIDR {
+func List(s []string) []*CIDR {
```
Comment (UmanShahzad) [inline lib/ipUtils/cidr.go:51]: Let's pick a better name for this now, e.g. `NewCidrList`

**Review by UmanShahzad — COMMENTED** [review]
Looks fairly clean, just fix the naming to be `iputil` (lowercased and without suffix `s`), and review comments above, and Ill do one test spin

File: lib/ipUtils/cidr.go (hunk 2 above)
Comment (shamirShahzad) [reply lib/ipUtils/cidr.go]: Many of them are already capital

Comment (shamirShahzad) [reply lib/ipUtils/cidr.go]: The ones which were remaining I have changed them

**Comment by shamirShahzad** [comment]
@UmanShahzad I did all the things kindly check if I missed anything

**Review by UmanShahzad — APPROVED** [review]

**Pull request merged**